geode-pulse/src/main/java/org/apache/geode/tools/pulse/internal/security/OauthSecurityConfig.java

```diff
@@ -0,0 +1,100 @@
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.tools.pulse.internal.security;
+
+import static org.springframework.security.config.Customizer.withDefaults;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.oauth2.client.InMemoryOAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.client.registration.InMemoryClientRegistrationRepository;
+import org.springframework.security.oauth2.client.web.AuthenticatedPrincipalOAuth2AuthorizedClientRepository;
+import org.springframework.security.oauth2.client.web.OAuth2AuthorizedClientRepository;
+import org.springframework.security.oauth2.core.AuthorizationGrantType;
+
+@Configuration
+@EnableWebSecurity
+@EnableGlobalMethodSecurity(prePostEnabled = true)
+@Profile("pulse.authentication.oauth")
+@PropertySource("classpath:pulse.properties")
+public class OauthSecurityConfig extends WebSecurityConfigurerAdapter {
+  @Value("${pulse.oauth.provider}")
+  private String providerId;
+  @Value("${pulse.oauth.clientId}")
+  private String clientId;
+  @Value("${pulse.oauth.clientSecret}")
+  private String clientSecret;
+  @Value("${pulse.oauth.authorizationUri}")
+  private String authorizationUri;
+  @Value("${pulse.oauth.tokenUri}")
+  private String tokenUri;
+  @Value("${pulse.oauth.userInfoUri}")
+  private String userInfoUri;
+  @Value("${pulse.oauth.jwkSetUri}")
+  private String jwkSetUri;
+  @Value("${pulse.oauth.userNameAttributeName}")
+  private String userNameAttributeName;
+
+  @Override
+  protected void configure(HttpSecurity http) throws Exception {
+    http.authorizeRequests(authorize -> authorize
+        .anyRequest().authenticated())
```

Comment:
Do we need to add something here to take of where the app goes after logging out? For OAuth this should be different from Default or Gemfire flows I believe.

Reply:
In your test, what does it do?

Reply:
> Does this PR need to update some of the doc files to describe how to configure pulse to use oauth2?

Yes, I will add a task to the tracker story to update the doc. PR for that is to be followed.

Reply:
> In your test, what does it do?

Don't have that covered in test yet but it follows the `logoutHandler` set up in the `DefaultSecurityConfig`. In the OAuth case I think it should invalidate the token someway. I assume there is a built-in way that Spring handles this so we should try to make use of that.

Reply:
In this changeset, I have OauthSecurityConfig directly extends `WebSecurityConfigurerAdapter`. It does not use anything defined in DefaultSecurityConfig. So I hope spring would pick up the flow.